gradle-jdks-certs/src/main/resources/gradle-jdk-resolver.sh

```diff
@@ -0,0 +1,128 @@
+# limitations under the License.
+#
+
+##############################################################################
+#
+#   Gradle jdk set up script for POSIX generated by gradle-jdks.
+#
+#   This script does the following:
+#   (1) Downloads the corresponding JDK distribution based on the majorVersion=`gradle/gradle-jdk-major-version.txt`
+#     and the distribution_url=`gradle/jdks/${majorVersion}/${os}/${arch}/download_url.txt`
+#   (2) Installs the distribution in `$HOME/.gradle/gradle-jdks/${local_path}` based on the
+#     local_path=`gradle/jdks/${majorVersion}/${os}/${arch}/local_path.txt`
+#   (3) Calls the java class `JdkSpecCertSetup` that will set up the certificates for the locally installed distribution
+#   (4) Sets up the JAVA_HOME env variable to the currently installed JDK
+#
+#
+#   Important for running:
+#   This script requires all of these POSIX shell features:
+#         * functions;
+#         * expansions «$var», «${var}», «${var%suffix}», and «$( cmd )»;
+#         * various built-in commands including «command» and «set».
+#
+##############################################################################
+
+tmp_work_dir=$(mktemp -d)
+
+die () {
+    echo
+    echo "$*"
+    echo
+    # TODO(crogoz): Cleaning up
+    echo rm -rf "$tmp_work_dir"
+    exit 1
+} >&2
+
+# OS specific support
+case "$( uname )" in                          #(
+  Linux* )          os_name="linux"  ;;       #(
+  Darwin* )         os_name="macos"  ;;       #(
+  * )               die "ERROR Unsupported OS: $( uname )" ;;
+esac
+
+if [ "$os_name" = "linux" ]; then
+    if [ -f /etc/alpine-release ]; then
+      os_name="alpine-linux"
+    fi
+fi
+
+# Arch specific support, see: gradle-jdks:com.palantir.gradle.jdks.CurrentArch.java
+case "$(uname -m)" in                         #(
+  x86-64* )       arch_name="x86-64"  ;;      #(
+  x64* )          arch_name="x86-64"  ;;      #(
+  amd64* )        arch_name="x86-64"  ;;      #(
+  arm64* )        arch_name="aarch64"  ;;     #(
+  arm* )          arch_name="aarch64"  ;;     #(
+  aarch64* )      arch_name="aarch64"  ;;     #(
+  x86* )          arch_name="x86"  ;;         #(
+  i686* )         arch_name="x86"  ;;         #(
+  * )             die "ERROR Unsupported architecture: $( uname -m )" ;;
+esac
+
+read -r major_version < gradle/gradle-jdk-major-version.txt
+read -r distribution_url < gradle/jdks/"$major_version"/$os_name/$arch_name/download-url.txt
+read -r distribution_local_path < gradle/jdks/"$major_version"/$os_name/$arch_name/local-path.txt
+
+regex_pattern="^([a-zA-Z0-9_-]+)-([0-9.]+)-([a-z0-9]+)$"
+
+# Check if the input string matches the regex pattern
+if echo "$distribution_local_path" | grep -q -E "$regex_pattern"; then
+  name=$(echo "$distribution_local_path" | sed -n -E "s/$regex_pattern/\1/p")
+  version=$(echo "$distribution_local_path" | sed -n -E "s/$regex_pattern/\2/p")
+  hash=$(echo "$distribution_local_path" | sed -n -E "s/$regex_pattern/\3/p")
```

Comment:
I don't think we use `name`/`version`/`hash` anywhere other than putting it back together on line 91 - can we avoid parsing here at all? `sed` often has differences depending on if users have BSD or GNU sed installed. Would also be great to avoid needing grep.

Reply:
Oh, I see `$name-$version-inprogress`. I would honestly think it's better to just put all of these in separate files if we need to in the repo rather than do parsing in bash (basically move the complexity to the gradle side)

Reply:
you are right, will remove this

Reply:
having the hash as well is actually probably better.